lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -223,24 +443,13 @@ def filter_query(
         and_list = []
         for kw in filters:
             combine_or = False
-            contains = False
-            try:
-                k, v = kw.split(":", maxsplit=1)
-            except ValueError:
-                raise APIAbort(
-                    HTTPStatus.BAD_REQUEST, f"filter {kw!r} must have the form 'k:v'"
-                )
-            if k.startswith("^"):
+            term = Term(kw).parse_filter()
+            if term.chain == "^":
                 combine_or = True
-                k = k[1:]
-            if v.startswith("~"):
-                contains = True
-                v = v[1:]
-
-            if not Metadata.is_key_path(k, Metadata.METADATA_KEYS, metalog_key_ok=True):
-                raise APIAbort(HTTPStatus.BAD_REQUEST, str(MetadataBadKey(k)))
-            keys = k.split(".")
+            keys = term.key.split(".")
             native_key = keys.pop(0).lower()
+            vtype = term.type if term.type else "str"
```

Comment:
Consider initializing `type` to `"str"` in the `Term` c'tor.

Reply:
I *should* add default values in the constructor, unfortunately; but I rather like the ternary here as otherwise it goes to two lines for a conventional `if`.